lib/mariaex/protocol.ex

```diff
@@ -335,6 +348,7 @@ defmodule Mariaex.Protocol do
     end
   end
   def handle_prepare(%Query{type: :binary} = query, opts, %{binary_as: binary_as} = s) do
+    query = sanitize_query(query, s)
```

Comment:
Should be done in `prepare_insert`

Reply:
Please read https://github.com/xerions/mariaex/pull/243#discussion_r228348565

Also, I am not sure how to solve this issue by altering the method `prepare_insert`. Do you have a suggestion?

Reply:
Unfortunately, the code is not that easy to follow in order for me to fix the issue by altering just `prepare_insert`. I cannot quite get along with both LruCache and Cache to get the desired behaviour.

Can you give me some pointers maybe?

Reply:
Hey, James. I have cleaned up some unnecessary code. I am deleting the LRU cache when prepare == :unnamed which solved the problem of not being able to use `""` as unnamed query name.

The final problem: I cannot figure out how to stabilize the prepare statements count by moving the query override from `handle_prepare` to `prepare_insert`. I guess this is as far as I can go regarding complying to your feedback.

Hopefully you can help me out on this.